src/components/EditCardPaymentForm/EditCardPaymentForm.tsx

```diff
@@ -0,0 +1,130 @@
+import { object, string } from 'yup';
+import { useMutation } from 'react-query';
+import shallow from 'zustand/shallow';
+
+import Button from '../Button/Button';
+import CreditCardCVCField from '../CreditCardCVCField/CreditCardCVCField';
+import CreditCardExpiryField from '../CreditCardExpiryField/CreditCardExpiryField';
+import CreditCardNumberField from '../CreditCardNumberField/CreditCardNumberField';
+import TextField from '../TextField/TextField';
+
+import styles from './EditCardPaymentForm.module.scss';
+
+import useForm from '#src/hooks/useForm';
+import { updateCardDetails } from '#src/stores/AccountController';
+import { useAccountStore } from '#src/stores/AccountStore';
+
+type Props = {
+  onCancel: () => void;
+  setUpdatingCardDetails: (e: boolean) => void;
+};
+
+const EditCardPaymentForm: React.FC<Props> = ({ onCancel, setUpdatingCardDetails }) => {
+  const { t } = useTranslation('account');
+  const updateCard = useMutation(updateCardDetails);
+  const { activePayment } = useAccountStore(({ activePayment }) => ({ activePayment }), shallow);
+  const paymentData = useForm(
+    { cardholderName: '', cardNumber: '', cardExpiry: '', cardCVC: '', cardExpMonth: '', cardExpYear: '' },
+    async () => {
+      setUpdatingCardDetails(true);
+      updateCard.mutate(
+        {
+          cardName: paymentData.values.cardholderName,
+          cardNumber: paymentData.values.cardNumber.replace(/\s+/g, ''),
+          cvc: parseInt(paymentData.values.cardCVC),
+          expMonth: parseInt(paymentData.values.cardExpMonth),
+          expYear: parseInt(paymentData.values.cardExpYear),
+          currency: activePayment?.currency || '',
+        },
+        {
+          onSettled: () => onCancel(),
+        },
+      );
+    },
+    object().shape({
+      cardNumber: string().test('card number validation', t('checkout.invalid_card_number'), (value) => {
+        return Payment.fns.validateCardNumber(value as string);
+      }),
+      cardExpiry: string().test('card expiry validation', t('checkout.invalid_card_expiry'), (value) => {
+        return Payment.fns.validateCardExpiry(value as string);
+      }),
+      cardCVC: string().test('cvc validation', t('checkout.invalid_cvc_number'), (value) => {
+        const issuer = Payment.fns.cardType(paymentData?.values?.cardNumber);
+        return Payment.fns.validateCardCVC(value as string, issuer);
+      }),
+    }),
+
+    true,
+  );
+
+  useEffect(() => {
+    if (paymentData.values.cardExpiry) {
+      const expiry = Payment.fns.cardExpiryVal(paymentData.values.cardExpiry);
+      if (expiry.month) {
+        paymentData.setValue('cardExpMonth', expiry.month.toString());
+      }
+      if (expiry.year) {
+        paymentData.setValue('cardExpYear', expiry.year.toString());
+      }
+    }
+    //eslint-disable-next-line
+  }, [paymentData.values.cardExpiry]);
+
+  return (
+    <div className={styles.paymentForm}>
+      <div>
+        <TextField
+          label="Cardholder name"
```

Comment:
We should translate this field as well

Reply:
935748a0862d1a60e85231924c272b4981f42060